lib/src/lifecycle_module.dart

```diff
@@ -49,7 +49,8 @@ enum LifecycleState {
 
 /// Intended to be extended by most base module classes in order to provide a
 /// unified lifecycle API.
-abstract class LifecycleModule extends Object with Disposable {
+abstract class LifecycleModule implements DisposableManager {
+  final Disposable _disposable = new Disposable();
```

Comment:
Maybe `_disposableProxy` or something? I dunno, if other people are fine with the name I'm down.